tests/dialects/test_memref.py

```diff
@@ -291,8 +310,12 @@ def test_memref_cast():
 
 
 def test_dma_start():
-    src_type = MemRefType.from_element_type_and_shape(i64, [4, 512], memory_space=1)
```

Comment:
@AntonLydike I'm not sure what you meant here originally

Reply:
I don't understand, what do you mean?

Reply:
I now understand that there's a check that the memory spaces aren't the same, it's jus that the type here is an attribute, not int, but my change seems to achieve the same result of being nonequal attributes